tests/ci/ci.py

```diff
@@ -1873,26 +1875,27 @@ def _run_test(job_name: str, run_command: str) -> int:
         print("Use run command from a job config")
     else:
         print("Use run command from the workflow")
-    os.environ["CHECK_NAME"] = job_name
+    env["CHECK_NAME"] = job_name
     print(f"Going to start run command [{run_command}]")
-    process = subprocess.run(
-        run_command,
-        stdout=sys.stdout,
-        stderr=sys.stderr,
-        text=True,
-        check=False,
-        shell=True,
-    )
+    stopwatch = Stopwatch()
+    job_log = Path(TEMP_PATH) / "job_log.txt"
+    with TeePopen(run_command, job_log, env, timeout) as process:
+        retcode = process.wait()
+        if retcode != 0:
+            print(f"Run action failed for: [{job_name}] with exit code [{retcode}]")
+            if timeout and process.timeout_exceeded:
+                print(f"Timeout {timeout} exceeded, dumping the job report")
+                JobReport(
```

Comment:
Currently all CI job scripts:
1. do the work
2. create JobReport for further processing by ci.py script

You added a new case where it's not true any more, ci job may be terminated and job report will be created from ci.py. Is there any strong reason to create this corner case? Why can't we do the same from the job scripts?

Reply:
The reason is that here is a single place to handle timeouts.

We need not the statuses, but [test_name=Check timeout expired](https://play.clickhouse.com/play?user=play#U0VMRUNUICogRlJPTSBjaGVja3MgV0hFUkUgdGVzdF9uYW1lID09ICdDaGVjayB0aW1lb3V0IGV4cGlyZWQnIEFORCBjaGVja19zdGFydF90aW1lIEJFVFdFRU4gdG9EYXRlKCcyMDI0LTA1LTEwJykgQU5EIHRvRGF0ZSgnMjAyNC0wNS0xNScp) in the CI DB.

Now, every script having a timeout in ci_config implements the process around it like animals in a zoo. One process env, another argument. And the third one is in damn docker.

We need to have a single place for it. Maybe there's a better place where to do it. But we need to do it only once.

Reply:
https://play.clickhouse.com/play?user=play#U0VMRUNUICogRlJPTSBjaGVja3MgV0hFUkUgdGVzdF9uYW1lID09ICdDaGVjayB0aW1lb3V0IGV4cGlyZWQnIEFORCB0b0RhdGUoY2hlY2tfc3RhcnRfdGltZSkgPT0gdG9EYXRlKCcyMDI0LTA1LTE2Jyk=

That's what we need, + https://s3.amazonaws.com/clickhouse-test-reports/63982/03a1487a30cc52ae0aebde0fcb7ed48c7dc1b348/style_check.html

Reply:
ok. I agree, it makes sense to control execution time from ci.py

I see we have timeout inside stress_test and fast_test - it should be aligned with this new approach

What do you think about reporting timeout not as special/virtual _test_name_ but as a _check_status_ ?

Reply:
Check status is **mostly** [equal](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX3N0YXR1cyBGUk9NIGNoZWNrcyBXSEVSRSAgdG9EYXRlKGNoZWNrX3N0YXJ0X3RpbWUpID09IHRvRGF0ZSgnMjAyNC0wNS0xNicpIEdST1VQIEJZIEFMTA==) to valid states of commit status.

I'd rather fix [install_check.py](https://play.clickhouse.com/play?user=play#U0VMRUNUICogRlJPTSBjaGVja3MgV0hFUkUgY2hlY2tfc3RhdHVzID09ICdPSycgQU5EIHRvRGF0ZShjaGVja19zdGFydF90aW1lKSA+PSB0b0RhdGUoJzIwMjQtMDUtMTAnKSBHUk9VUCBCWSBBTEwgTElNSVQgMTA=) and have this completely equal.

Reply:
check_status is just a right place for that, even if github does not have such value

Reply:
But what will be the test_name?

Reply:
I've grep for `check_status`. Not sure if I follow the idea.

Can you explain?

Reply:
no test case - no test_name, right?

Reply:
I don't see a point. The state is used in GitHub commit status. It's a set of 4 valid strings.

Reply:
Agreed to disagree and keep `Check timeout expired` to not play around non-standard status state `timeout`